streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -214,6 +215,20 @@ public RocksDBMetricsRecordingTrigger rocksDBMetricsRecordingTrigger() {
         }
     }
 
+    public final Sensor clientLevelSensor(final String sensorName,
+                                          final RecordingLevel recordingLevel,
+                                          final Sensor... parents) {
+        final String key = CLIENT_LEVEL_GROUP;
+        synchronized (clientLevelSensors) {
+            final String fullSensorName = key + SENSOR_NAME_DELIMITER + sensorName;
+            return Optional.ofNullable(metrics.getSensor(fullSensorName))
+                .orElseGet(() -> {
+                    clientLevelSensors.push(fullSensorName);
+                    return metrics.sensor(fullSensorName, recordingLevel, parents);
+                });
```

Comment:
Although, we use this in other methods, I think the following is a bit simpler to read:
```suggestion
        final Sensor sensor = metrics.getSensor(fullSensorName);
        if (sensor == null) {
            clientLevelSensors.push(fullSensorName);
            return metrics.sensor(fullSensorName, recordingLevel, parents);
        }
        return sensor;
```